docs/hazmat/primitives/asymmetric/dh.rst

```diff
@@ -5,6 +5,11 @@ Diffie-Hellman key exchange
 
 .. currentmodule:: cryptography.hazmat.primitives.asymmetric.dh
 
+.. note::
+    We suggest using
+    :class:`~cryptography.hazmat.primitives.asymmetric.ec.ECDH` instead of DH
+    where possible.
```

Comment:
I guess compatibility cuts both ways, but security and performance seems reasonable.